resources/dw/virtualwarehouse/hive/model_hive_vw.go

```diff
@@ -11,21 +11,151 @@
+	JwtConnectionString          types.String          `tfsdk:"jwt_connection_string"`
+	JwtTokenGenUrl               types.String          `tfsdk:"jwt_token_gen_url"`
+	MinGroupCount                types.Int64           `tfsdk:"min_group_count"`
+	MaxGroupCount                types.Int64           `tfsdk:"max_group_count"`
+	DisableAutoSuspend           types.Bool            `tfsdk:"disable_auto_suspend"`
+	AutoSuspendTimeoutSeconds    types.Int64           `tfsdk:"auto_suspend_timeout_seconds"`
+	ScaleWaitTimeSeconds         types.Int64           `tfsdk:"scale_wait_time_seconds"`
+	Headroom                     types.Int64           `tfsdk:"headroom"`
+	MaxConcurrentIsolatedQueries types.Int64           `tfsdk:"max_concurrent_isolated_queries"`
+	MaxNodesPerIsolatedQuery     types.Int64           `tfsdk:"max_nodes_per_isolated_query"`
+	AwsOptions                   *awsOptions           `tfsdk:"aws_options"`
+	LastUpdated                  types.String          `tfsdk:"last_updated"`
+	Status                       types.String          `tfsdk:"status"`
+	PollingOptions               *utils.PollingOptions `tfsdk:"polling_options"`
 }
 
 func (p *resourceModel) GetPollingOptions() *utils.PollingOptions {
 	return p.PollingOptions
 }
+
+func (p *resourceModel) convertToCreateVwRequest() *models.CreateVwRequest {
+	vwType := models.VwType("hive")
+	return &models.CreateVwRequest{
+		ClusterID:             p.ClusterID.ValueStringPointer(),
+		DbcID:                 p.DatabaseCatalogID.ValueStringPointer(),
+		EbsLLAPSpillGB:        p.getEbsLLAPSpillGB(),
+		ImageVersion:          p.getImageVersion(),
+		Name:                  p.Name.ValueStringPointer(),
+		NodeCount:             utils.Int64To32(p.GroupSize),
+		PlatformJwtAuth:       p.PlatformJwtAuth.ValueBoolPointer(),
+		QueryIsolationOptions: p.getQueryIsolationOptions(),
+		Autoscaling:           p.getAutoscaling(),
+		AvailabilityZone:      p.getAvailabilityZone(),
+		Tags:                  p.getTags(),
+		Config:                p.getServiceConfig(),
+		VwType:                &vwType,
+	}
+}
+
+func (p *resourceModel) getImageVersion() string {
+	if p.ImageVersion.IsNull() || p.ImageVersion.IsUnknown() || p.ImageVersion.String() == "<unknown>" {
+		return ""
+	}
+	return strings.TrimPrefix(strings.TrimSuffix(p.ImageVersion.String(), "\""), "\"")
+}
+
+func (p *resourceModel) getServiceConfig() *models.ServiceConfigReq {
+	return &models.ServiceConfigReq{
+		ApplicationConfigs: nil,
+		CommonConfigs:      nil,
+		EnableSSO:          p.EnableSSO.ValueBool(),
+		LdapGroups:         utils.FromListValueToStringList(p.LdapGroups),
+	}
+}
+
+func (p *resourceModel) getTags() []*models.TagRequest {
+	if p.AwsOptions.Tags.IsNull() {
+		return nil
+	}
+	var tags []*models.TagRequest
+	for k, v := range p.AwsOptions.Tags.Elements() {
+		if v.IsNull() {
+			continue
+		}
+		value := strings.TrimPrefix(strings.TrimSuffix(v.String(), "\""), "\"")
```

Comment:
Why is this sanitization required?

Reply:
it won't be required, in the next patch I will remove this and use StringValue where applicable